pkg/storage/ingest/version.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingest
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+)
+
+const (
+	RecordVersionHeaderKey = "X-Record-Version"
```

Comment:
[nit] Have you considered a shorter string, given this is going to be encoded in each record?

Reply:
Fair enough. I'll move to just `Version` then. No need to use HTTP semantics and it's obviously a record :smile: